### ISSUE TITLE
Analyzer Feedback: NoSuchMethodError: The getter 'isDynamic' was called on null

### ISSUE THREAD

**Issue by MikeMitterer** [dart-lang/sdk#28578]
[report.txt](https://github.com/dart-lang/sdk/files/742638/report.txt)
Analyzer Feedback from IntelliJ

## Version information

- `IDEA 171.2613.7`
- `1.21.1`
- `IU-171.2613.7, JRE 1.8.0_112-release-b696x64 JetBrains s.r.o, OS Mac OS X(x86_64) v10.11.6 unknown, screens 1920x1080, 1920x1080, 2560x1440`

## Exception

Please append the contents of:
file:///private/var/folders/m4/9rg16xhd2rlbln753n9zk1540000gn/T/report.txt

```
Dart analysis server, SDK version 1.21.1, server version 1.17.0, error: Task failed: StrongModeVerifyUnitTask for source /Volumes/Daten/DevLocal/DevDart/MobiAd.REST.UI.MDL/test/integration/services/LoginService_test.dart
Unexpected exception while performing StrongModeVerifyUnitTask for source /Volumes/Daten/DevLocal/DevDart/MobiAd.REST.UI.MDL/test/integration/services/LoginService_test.dart
#0      AnalysisTask._safelyPerform (package:analyzer/task/model.dart:333)
#1      AnalysisTask.perform (package:analyzer/task/model.dart:220)
#2      AnalysisDriver.performWorkItem (package:analyzer/src/task/driver.dart:284)
#3      AnalysisDriver.performAnalysisTask (package:analyzer/src/task/driver.dart:255)
#4      AnalysisContextImpl.performAnalysisTask.<anonymous closure> (package:analyzer/src/context/context.dart:1186)
#5      _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:189)
#6      AnalysisContextImpl.performAnalysisTask (package:analyzer/src/context/context.dart:1184)
#7      PerformAnalysisOperation.perform (package:analysis_server/src/operation/operation_analysis.dart:362)
#8      AnalysisServer.performOperation (package:analysis_server/src/analysis_server.dart:972)
#9      Future.Future.<anonymous closure> (dart:async/future.dart:118)
#10     _rootRun (dart:async/zone.dart:1146)
#11     _CustomZone.run (dart:async/zone.dart:1026)
#12     _CustomZone.runGuarded (dart:async/zone.dart:924)
#13     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#14     _rootRun (dart:async/zone.dart:1150)
#15     _CustomZone.run (dart:async/zone.dart:1026)
#16     _CustomZone.runGuarded (dart:async/zone.dart:924)
#17     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:951)
#18     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:16)
#19     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:385)
#20     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:414)
#21     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:148)

Caused by NoSuchMethodError: The getter 'isDynamic' was called on null.
Receiver: null
Tried calling: isDynamic
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)
#2      CodeChecker._checkForUnsafeBlockClosureInference (package:analyzer/src/task/strong/checker.dart:789)
#3      CodeChecker.visitFunctionExpression (package:analyzer/src/task/strong/checker.dart:438)
#4      FunctionExpressionImpl.accept (package:analyzer/src/dart/ast/ast.dart:5330)
#5      NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7814)
#6      ArgumentListImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:431)
#7      RecursiveAstVisitor.visitArgumentList (package:analyzer/dart/ast/visitor.dart:596)
#8      ArgumentListImpl.accept (package:analyzer/src/dart/ast/ast.dart:427)
#9      MethodInvocationImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:7426)
#10     CodeChecker.visitMethodInvocation (package:analyzer/src/task/strong/checker.dart:573)
#11     MethodInvocationImpl.accept (package:analyzer/src/dart/ast/ast.dart:7419)
#12     ExpressionStatementImpl.visitChildren (package:analyzer/src/dart/ast/ast.dart:4257)
#13     RecursiveAstVisitor.visitExpressionStatement (package:analyzer/dart/ast/visitor.dart:806)
#14     ExpressionStatementImpl.accept (package:analyzer/src/dart/ast/ast.dart:4253)
#15     NodeListImpl.accept (package:analyzer/src/dart/ast/ast.dart:7814)

**Comment by bwilkerson** [comment]
@scheglov Is this the same as the bug you fixed yesterday?

**Comment by scheglov** [comment]
I cannot find confirmation of this.
The bug I fixed yesterday has a different stack trace.
So, I guess this is a different bug.

Too bad crash dumps are implemented only after 1.22 RC and only when the new analysis driver is enabled :-(

```
Caused by NoSuchMethodError: The getter 'isDynamic' was called on null.
Receiver: null
Tried calling: isDynamic
#0      Object._noSuchMethod (dart:core-patch/object_patch.dart:44)
#1      Object.noSuchMethod (dart:core-patch/object_patch.dart:47)
#2      InterfaceTypeImpl.isMoreSpecificThan (package:analyzer/src/dart/element/type.dart:1673:14)
#3      ErrorVerifier._checkForTypeParameterSupertypeOfItsBound (package:analyzer/src/generated/error_verifier.dart:5794:16)
#4      ErrorVerifier.visitTypeParameter (package:analyzer/src/generated/error_verifier.dart:1231:5)```

**Comment by devoncarew** [comment]
Thanks for the report! It looks like the stack trace for this exception is significantly different from the current code that we're unlikely to be able to repro.
